packages/java-spring/test/components/spring-service-endpoint.test.tsx

```diff
@@ -0,0 +1,315 @@
+      <jv.Class name={"TestClass"}>
+        <SpringServiceEndpoint op={Bar} />
+      </jv.Class>
+    </>  });
+
+  expect(output).toBe(d`
+      package me.test.code;
+      
+      import org.springframework.web.bind.annotation.GetMapping;
+      import org.springframework.web.bind.annotation.RequestHeader;
+      
+      class TestClass {
+        @GetMapping("/people")
+        public String listPeople(@RequestHeader("name") String name);
+      }
+  `);
+});
+
+it("Creates service with path parameter", async () => {
+  const code = `
+    @route("/people")
+    op listPeople(@path name: string): void;
+  `;
+
+  const output = await getEmitOutput(code, (program) => {
+    const Foo = program.resolveTypeReference("listPeople")[0]! as Operation;
+    const Bar = $.httpOperation.get(Foo);
+    return <>
+      <jv.Class name={"TestClass"}>
+        <SpringServiceEndpoint op={Bar} />
+      </jv.Class>
+    </>  });
+
+  expect(output).toBe(d`
+      package me.test.code;
+      
+      import org.springframework.web.bind.annotation.GetMapping;
+      import org.springframework.web.bind.annotation.PathVariable;
+      
+      class TestClass {
+        @GetMapping("/people/{name}")
+        public void listPeople(@PathVariable("name") String name);
+      }
+  `);
+});
+
+it("Creates service with path parameter from the path", async () => {
+  const code = `
+    @route("/people/{name}")
+    op listPeople(name: string): void;
+  `;
+
+  const output = await getEmitOutput(code, (program) => {
+    const Foo = program.resolveTypeReference("listPeople")[0]! as Operation;
+
+    const Bar = $.httpOperation.get(Foo);
+    return <>
+      <jv.Class name={"TestClass"}>
+        <SpringServiceEndpoint op={Bar} />
+      </jv.Class>
+    </>  });
+
+  expect(output).toBe(d`
+      package me.test.code;
+      
+      import org.springframework.web.bind.annotation.GetMapping;
+      import org.springframework.web.bind.annotation.PathVariable;
+
+      class TestClass {
+        @GetMapping("/people/{name}")
+        public void listPeople(@PathVariable String name);
+      }
+  `);
+});
+
+it("Creates service with query parameter", async () => {
+  const code = `
+    @route("/people")
+    op listPeople(@query name: string): void;
+  `;
+
+  const output = await getEmitOutput(code, (program) => {
+    const Foo = program.resolveTypeReference("listPeople")[0]! as Operation;
+    const Bar = $.httpOperation.get(Foo);
+    return <>
+      <jv.Class name={"TestClass"}>
+        <SpringServiceEndpoint op={Bar} />
+      </jv.Class>
+    </>  });
+
+  expect(output).toBe(d`
+      package me.test.code;
+      
+      import org.springframework.web.bind.annotation.GetMapping;
+      import org.springframework.web.bind.annotation.RequestParam;
+
+      class TestClass {
+        @GetMapping("/people")
+        public void listPeople(@RequestParam("name") String name);
+      }
+  `);
+});
+
+it("Creates service with declared model parameter", async () => {
```

Comment:
Awesome work! I think take this specific test out for now since this is more an end-to-end test and not really a unit test for the component. Will need to do something slightly different in this test but will probably be easier if you remove this test for now and we can do it properly later.